src/Push.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <nanomsg/pipeline.h>
+#include "NanoMsg.h"
+#include "NanoProtocol.h"
+#include <memory>
+class Push {
+public:
+   Push(const std::string& location, const bool shouldConnect=true);
+   Push(const std::string&,
+        const int timeoutInMs,
+        const bool shouldConnect=true);
```

Comment:
@KjellKod how would you accomplish this then? this is a general problem with c++ i have. to me it is easier to take care of the default if it's not something the user is normally interested in.

push('location')
push('location', true)

is easier than
than push(location, false) being declared all the time.